pkg/controller/servicemesh/controlplane/htpasswd.go

```diff
@@ -56,6 +67,29 @@ func (r *controlPlaneInstanceReconciler) patchHtpasswdSecret(ctx context.Context
 	return nil
 }
 
+func hashPassword(version versions.Version, rawPass string) (string, error) {
+	var auth, hashedPassword string
+	username := "internal"
+
+	// If the SMCP version is below v2.4, we use SHA-1 to keep behavior of old SMCP versions consistent,
+	// and to keep oauth-proxy version consistent at version 4.4 in those releases.
+	if versions.V2_4.Compare(version) > 0 {
```

Comment:
```suggestion
	if version.AtLeast(V2_4) {
```
And then move code from `else` branch to this condition.

Reply:
made change in [7bd48fc](https://github.com/maistra/istio-operator/pull/1161/commits/7bd48fc092911eb7f8746dc3c724debaab355351) :+1: